boa/src/syntax/parser/expression/assignment/arrow_function.rs

```diff
@@ -127,14 +133,16 @@ where
         {
             let lexically_declared_names = body.lexically_declared_names();
             for param in params.parameters.as_ref() {
-                if lexically_declared_names.contains(param.name()) {
-                    return Err(ParseError::lex(LexError::Syntax(
-                        format!("Redeclaration of formal parameter `{}`", param.name()).into(),
-                        match cursor.peek(0)? {
-                            Some(token) => token.span().end(),
-                            None => Position::new(1, 1),
-                        },
-                    )));
+                for param_name in param.names().iter() {
```

Comment:
```suggestion
                for param_name in param.names() {
```
The `iter` call is unnecessary in a lot of these, the `for` expression automatically calls `into_iter` if you pass something implementing the `IntoIterator` trait.